From ebc164ef40cfa0fa01ce77a6a966129cc611d4ff Mon Sep 17 00:00:00 2001
From: Tom Gundersen <teg@jklm.no>
Date: Mon, 15 Sep 2014 11:53:03 +0200
Subject: [PATCH] udev: only print after final log level has been determined

This delays label_init(), and drops the (duplicate) printing of version
information.
---
 src/udev/udevd.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/udev/udevd.c b/src/udev/udevd.c
index 04014b468a..cfa071eba9 100644
--- a/src/udev/udevd.c
+++ b/src/udev/udevd.c
@@ -1104,9 +1104,6 @@ int main(int argc, char *argv[]) {
         udev_set_log_fn(udev, udev_main_log);
         log_set_max_level(udev_get_log_priority(udev));
 
-        log_debug("version %s", VERSION);
-        label_init("/dev");
-
         r = parse_argv(argc, argv);
         if (r <= 0)
                 goto exit;
@@ -1123,6 +1120,8 @@ int main(int argc, char *argv[]) {
                 goto exit;
         }
 
+        label_init("/dev");
+
         /* set umask before creating any file/directory */
         chdir("/");
         umask(022);
